code/modules/projectiles/gun_attachables/rail.dm

```diff
@@ -145,20 +145,6 @@
 	. = ..()
 	ENABLE_BITFIELD(master_gun.item_flags, IS_DEPLOYABLE)
 	master_gun.deployable_item = /obj/machinery/deployable/mounted/sentry/buildasentry
-	master_gun.ignored_terrains = list(
-		/obj/machinery/deployable/mounted,
-		/obj/machinery/miner,
-	)
-	if(master_gun.ammo_datum_type && CHECK_BITFIELD(initial(master_gun.ammo_datum_type.ammo_behavior_flags), AMMO_ENERGY) || istype(master_gun, /obj/item/weapon/gun/energy)) //If the guns ammo is energy, the sentry will shoot at things past windows.
```

Comment:
Why is this behaviour being taken away? If anything you could just make all sentries ignore windows.